packages/router/src/route_reuse_strategy.ts

```diff
@@ -60,20 +60,55 @@ export abstract class RouteReuseStrategy {
 }
 
 /**
- * Does not detach any subtrees. Reuses routes as long as their route config is the same.
+ * @description
+ *
+ * This base route reuse strategy only reuses routes when the matched router configs are
+ * identical. This prevents components from being destroyed and recreated
+ * when the just, the fragment, or query parameters change
+ * or when `ExtraOptions#onSameUrlNavigation` is set to 'reload'
+ * (that is, the existing component is _reused_).
+ *
+ * This strategy does not store any routes for later reuse.
+ *
+ * Angular uses this strategy by default.
+ *
+ *
+ * Extension friendly
```

Comment:
I think it'd be great to rephrase it a bit to be specific what extension means.
```suggestion
 * It can be used as a base class for custom route reuse strategies, i.e. you can create your own class that extends the `BaseRouteReuseStrategy` one.
```